core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -877,8 +889,17 @@ class DAGScheduler(
       // SparkListenerStageCompleted here in case there are no tasks to run.
       outputCommitCoordinator.stageEnd(stage.id)
       listenerBus.post(SparkListenerStageCompleted(stage.latestInfo))
-      logDebug("Stage " + stage + " is actually done; %b %d %d".format(
-        stage.isAvailable, stage.numAvailableOutputs, stage.numPartitions))
+
+      val debugString = stage match {
+        case stage: ShuffleMapStage =>
+          "Stage " + stage + (" is actually done; " +
+            s"(available: ${stage.isAvailable}," +
+            s"available outputs: ${stage.numAvailableOutputs}," +
+            s"partitions: ${stage.numPartitions})")
+        case stage : ResultStage =>
+          "Stage " + stage + s" is actually done; (partitions: ${stage.numPartitions})"
```

Comment:
can you just do string interpolation all the way instead of a mixture of both? same in L895